src/session_impl.cpp

```diff
@@ -1082,7 +1082,7 @@ namespace aux {
 				// them consistent and unique per torrent and interface
 				req.key ^= ls->tracker_key;
 				req.outgoing_socket = ls;
-				m_tracker_manager.queue_request(get_io_service(), req, c);
+				m_tracker_manager.queue_request(get_io_service(), std::move(req), c);
```

Comment:
This is repeatedly moving `req` in a loop, I don't think this is safe.

Reply:
in fact, the logic for setting `key` isn't right either. It isn't supposed to include the keys from the trackers earlier in the loop